uniflow/op/model/neuron_utils.py

```diff
@@ -0,0 +1,297 @@
+            raise ValueError("MD5 verification failed")
+        print(zip_filename, os.path.join(cache_dir, f"{expected_md5}"))
+        with zipfile.ZipFile(zip_filename, "r") as zip_ref:
+            zip_ref.extractall(os.path.join(cache_dir, f"{expected_md5}"))
+        os.remove(zip_filename)
+        return os.path.join(cache_dir, f"{expected_md5}")
+
+    @staticmethod
+    def split_neuron_model(model_name: str) -> str:
+        """
+        Split weights
+
+        Args:
+            model_name (str): The name of the model to split weights from.
+
+        Returns:
+            str: The directory path where the split weights are saved.
+        """
+
+        cache_dir = os.path.join(
+            os.path.expanduser("~"),
+            ".cache/uniflow/",
+            model_name.split("/")[1] + "-split",
+        )
+        if not os.path.exists(cache_dir):
+            os.makedirs(cache_dir)
+        model_cpu = AutoModelForCausalLM.from_pretrained(model_name)
+        save_pretrained_split(model_cpu, cache_dir)
+        return cache_dir
+
+    @staticmethod
+    def get_neuron_model(model_name: str, batch_size: int):
+        """
+        Get the neuron model for the given model name and batch size.
+
+        Args:
+            model_name (str): The name of the model.
+            batch_size (int): The batch size.
+
+        Returns:
+            The neuron model.
+        """
+        instance_type = Neuron.get_instance_type()
+        assert "inf2" in instance_type, ValueError(
+            "Neuron model can only run on the AWS EC2 inf2 instance series."
+        )
+        tp_degree = Neuron.instance_type_dict[instance_type]
+        assert model_name in Neuron.neuron_model_config, ValueError(
+            f"Current Neuron only support {list(Neuron.neuron_model_config.keys())}"
+        )
+        assert batch_size in Neuron.neuron_model_config[model_name], ValueError(
+            f"{model_name} only support batch size {list(Neuron.neuron_model_config[model_name].keys())} in {instance_type}"
+        )
+        n_positions_list = list(
+            Neuron.neuron_model_config[model_name][batch_size].keys()
+        )
+        n_positions_list.sort()
+        n_positions = max(Neuron.neuron_model_config[model_name][batch_size])
+        neuron_config = NeuronConfig(
+            grouped_query_attention=constants.GQA.SHARD_OVER_HEADS
+        )
+        cache_dir = os.path.join(
+            os.path.expanduser("~"),
+            ".cache/uniflow/",
+            model_name.split("/")[1] + "-split",
+        )
+        if os.path.exists(cache_dir):
+            print(
+                f"Model chunks found. If you encounter errors, please delete the cache folder `{cache_dir}`"
+            )
+            model_split = cache_dir
+        else:
+            print("Spliting model, need to wait for a while...")
+            model_split = Neuron.split_neuron_model(model_name)
+
+        model_neuron = MistralForSampling.from_pretrained(
```

Comment:
nit: I assume this is for Mistral only, so I recommend to leave a TODO here. In the future, we have to change this for other non Mistral models after add their binary path at `neuron_model_config`.